consensus/core/src/commit.rs

```diff
@@ -543,30 +543,30 @@ impl CommitInfo {
 }
 
 /// CommitRange stores a range of CommitIndex. The range contains the start (inclusive)
-/// and end (exclusive) commit indices and can be ordered for use as the key of a table.
-#[derive(Clone, Default, Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub(crate) struct CommitRange(Range<CommitIndex>);
+/// and end (inclusive) commit indices and can be ordered for use as the key of a table.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub(crate) struct CommitRange(RangeInclusive<CommitIndex>);
 
 #[allow(unused)]
 impl CommitRange {
-    pub(crate) fn new(range: Range<CommitIndex>) -> Self {
+    pub(crate) fn new(range: RangeInclusive<CommitIndex>) -> Self {
         Self(range)
     }
 
     // Inclusive
     pub(crate) fn start(&self) -> CommitIndex {
-        self.0.start
+        *self.0.start()
     }
 
     // Exclusive
     pub(crate) fn end(&self) -> CommitIndex {
-        self.0.end
+        *self.0.end()
     }
 
     /// Check if the provided range is sequentially after this range with the same
     /// range length.
     pub(crate) fn is_next_range(&self, other: &Self) -> bool {
-        self.0.len() == other.0.len() && self.end() == other.start()
+        self.0.size_hint() == other.0.size_hint() && self.end() + 1 == other.start()
```

Comment:
Why did we need to check size again? Seems a bit unexpected?

Reply:
We want to ensure that the next range starts after this range and is of the same size. This is an assertion we want to check before we update the leader swap table. RangeInclusive does not have len() method so we had to use size_hint which checks the steps between start and end.

Reply:
It does not seem very intuitive that `is_next_range()` also expects the sizes are the same. What about just making sure `self.end() + 1 == other.start()` here, and do the size check at the callsite? Or move this function to leader schedule.